src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -28,23 +28,13 @@
 
 use Cookie;
 use PrestaShop\PrestaShop\Adapter\Configuration;
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Configuration\AbstractMultistoreConfiguration;
 
 /**
  * This class will provide Shop Preferences configuration.
  */
-class PreferencesConfiguration implements DataConfigurationInterface
+class PreferencesConfiguration extends AbstractMultistoreConfiguration
 {
-    /**
-     * @var Configuration
-     */
-    private $configuration;
-
-    public function __construct(Configuration $configuration)
-    {
-        $this->configuration = $configuration;
-    }
```

Comment:
TODO: Calls to getter methods in `getConfiguration` method should take into account the `ShopConstraint`